python/ray/tests/test_logging.py

```diff
@@ -874,6 +888,313 @@ def test_ray_does_not_break_makeRecord():
         logging.setLogRecordFactory(logging.LogRecord)
 
 
+def test_one_log_per_workerpool_worker_processes_reuse_log_file(shutdown_only):
+    ray.init(
+        num_cpus=1,
+        _system_config={
+            "one_log_per_workerpool_worker": True,
+            "enable_worker_prestart": False,
+        },
+    )
+
+    actor_log_line = "my actor remote func"
+
+    @ray.remote
+    class MyActor:
+        def act(self):
+            print(actor_log_line)
+
+    actor = MyActor.options(name="MyActor").remote()
+    ray.get(actor.act.remote())
+
+    # Kill actor, which kills the worker, so the next task creates a worker with
+    # the same index and uses the same file name.
```

Comment:
Shall we also add a stress tests that launches say 1k tasks with max_calls=1, and checks the logs are evenly spread across the N indices?